### PR TITLE
reformat tables, grammar and style issues in documentation-files

### LINKED ISSUE #232: fixing typos and other style-issues in documentation-files
**Problem description**
Currently the documentation-files, e.g., API-design-guideline.md has some broken internal references (e.g., inside the table of content).
Also some other issues were found like: 

- unformated tables
- too long sentences, which were hardly to read on the first try.
- missing articles
- some grammar issues

### PR DESCRIPTION
**Pull request by maxl2287**
#### What type of PR is this?

Add one of the following kinds:
* cleanup

#### What this PR does / why we need it:
This PR is a fixing of some style and reference-issues 

#### Which issue(s) this PR fixes:

Fixes #232

### CODE REVIEW

**Review by patrice-conil — DISMISSED** [review]
Not good enough in English to judge these subtle changes so LGTM

File: documentation/Linting-rules.md [hunk 1]
```diff
@@ -7,15 +7,15 @@
...
+With `extends: "spectral:oas"` ("oas" being shorthand for OpenAPI Specification) in the ruleset file to rules for OpenAPI v2 and v3.x, depending on the appropriate OpenAPI version being used (this is automatically detected through formats) is added to the final ruleset.
 
-All of the rules in this ruleset are described in [OpenAPI Rules](https://docs.stoplight.io/docs/spectral/4dec24461f3af-open-api-rules).
+All the rules in this ruleset are described in [OpenAPI Rules](https://docs.stoplight.io/docs/spectral/4dec24461f3af-open-api-rules).
 
 
-`extends: [[spectral:oas, off]]`  - this avoids running any rules from the extended ruleset as they are disabled. Each rule then can be [enabled individually](https://docs.stoplight.io/docs/spectral/0a73453054745-recommended-or-all#enabling-rules).
+`extends: [[spectral:oas, off]]`  - This avoids running any rules from the extended ruleset as they are disabled. Each rule then can be [enabled individually](https://docs.stoplight.io/docs/spectral/0a73453054745-recommended-or-all#enabling-rules).
```
Comment (PedroDiez) [inline documentation/Linting-rules.md:15]:
```suggestion
`extends: [[spectral:oas, off]]` - This avoids running any rules from the extended ruleset as they are disabled. Each rule then can be [enabled individually](https://docs.stoplight.io/docs/spectral/0a73453054745-recommended-or-all#enabling-rules).

```

**Review by PedroDiez — APPROVED** [review]
minor comment 

LGTM in advance

**Comment by maxl2287** [comment]
@PedroDiez updated your typo-hint and resolve the merge conflicts.
Please review again :)

File: documentation/API-Testing-Guidelines.md [hunk 2]
```diff
@@ -269,7 +269,7 @@ Typical response validations:
 - the response body complies with the OAS schema at "{oas_schema_json_path}"
 ```
 
-- additional constraints for optional or dependant properties in the response body, e.g.:
+- additional constraints for optional or independent properties in the response body, e.g.:
```
Comment (PedroDiez) [inline documentation/API-Testing-Guidelines.md:272]:
```suggestion
- additional constraints for optional or dependent properties in the response body, e.g.:
```

Comment (PedroDiez) [reply documentation/API-Testing-Guidelines.md:272]: It is dependant due to the use cases illustrated (clause "if"):

- the response property "{json_path}" only exists **if** property "{json_path}" exists
- the response property "{json_path}" only exists **if** property "{json_path}" is "{value}"

File: documentation/API-Testing-Guidelines.md [hunk 3]
```diff
@@ -125,27 +125,27 @@ For the Feature description, API name and version must be included. When the fea
 Feature: CAMARA Device location verification API, v0.2.0 - Operation verifyLocation
 ```
 
-### Environment variables <a name="environment-variables"></a>
+### Environment variables
 
-Commonly, some values to fill the request bodies will not be known in advance and cannot be specified as part of the feature file, as they will be specific for the test environment, and they will have to be provided by the implementation to the tester as a separate set of environment or configuration variables.
+Commonly, some values to fill the request bodies will not be known in advance and cannot be specified as part of the feature file, as they will be specific for the test environment, and they will have to be provided by the implementation of the tester as a separate set of environment or configuration variables.
```
Comment (PedroDiez) [inline documentation/API-Testing-Guidelines.md:130]:
```suggestion
Commonly, some values to fill the request bodies will not be known in advance and cannot be specified as part of the feature file, as they will be specific for the test environment, and they will have to be provided by the implementation to the tester as a separate set of environment or configuration variables.
```

Comment (PedroDiez) [reply documentation/API-Testing-Guidelines.md:130]: It is "to", because it is something to be provided to the person/entity that performs the testing

**Review by PedroDiez — COMMENTED** [review]
A couple of comments regarding API testing guidelines

**Review by PedroDiez — DISMISSED** [review]
LGTM

**Review by patrice-conil — APPROVED** [review]
LGTM

**Review by PedroDiez — APPROVED** [review]
LGTM

**Comment by rartych** [comment]
Merging as commonly approved (the last commits resolved conficts)

**Pull request merged**